core/src/test/java/nu/yona/server/messaging/service/FirebaseServiceTest.java

```diff
@@ -0,0 +1,184 @@
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.FilterType;
+import org.springframework.data.repository.Repository;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import com.google.firebase.messaging.FirebaseMessaging;
+import com.google.firebase.messaging.FirebaseMessagingException;
+
+import nu.yona.server.Translator;
+import nu.yona.server.device.service.DeviceService;
+import nu.yona.server.entities.EntityWithId;
+import nu.yona.server.entities.UserRepositoriesConfiguration;
+import nu.yona.server.exceptions.YonaException;
+import nu.yona.server.messaging.entities.SystemMessage;
+import nu.yona.server.properties.YonaProperties;
+import nu.yona.server.test.util.BaseSpringIntegrationTest;
+import nu.yona.server.test.util.JUnitUtil;
+import nu.yona.server.util.AsyncExecutor;
+
+@Configuration
+@ComponentScan(useDefaultFilters = false, basePackages = { "nu.yona.server.messaging.service",
+		"nu.yona.server.properties" }, includeFilters = {
+				@ComponentScan.Filter(pattern = "nu.yona.server.messaging.service.FirebaseService", type = FilterType.REGEX),
+				@ComponentScan.Filter(pattern = "nu.yona.server.properties.YonaProperties", type = FilterType.REGEX) })
+class FirebaseServiceTestConfiguration extends UserRepositoriesConfiguration
+{
+}
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = { FirebaseServiceTestConfiguration.class })
+public class FirebaseServiceTest extends BaseSpringIntegrationTest
+{
+	private static final Constructor<FirebaseMessagingException> firebaseMessagingConstructor = JUnitUtil
+			.getAccessibleConstructor(FirebaseMessagingException.class, String.class, String.class, Throwable.class);
+	private static final Field messageIdField = JUnitUtil.getAccessibleField(EntityWithId.class, "id");
+
+	@MockBean
+	private Translator mockTranslator;
+
+	@MockBean
+	private AsyncExecutor mockAsyncExecutor;
+
+	@MockBean
+	private DeviceService mockDeviceService;
+
+	@MockBean
+	private FirebaseMessaging mockFirebaseMessaging;
+
+	@Autowired
+	private YonaProperties yonaProperties;
+
+	@Autowired
+	private FirebaseService service;
+
+	@BeforeEach
+	public void setUp()
+	{
+		doAnswer(invocation -> {
+			exec(invocation.getArgument(1), invocation.getArgument(2));
+			return null;
+		}).when(mockAsyncExecutor).execAsync(any(), any(), any());
+
+		yonaProperties.getFirebase().setEnabled(true);
+	}
+
+	@Override
+	protected Map<Class<?>, Repository<?, ?>> getRepositories()
+	{
+		return new HashMap<>();
+	}
+
+	@Test
+	public void sendMessage_success_sendsFirebaseMessage() throws FirebaseMessagingException
+	{
+		UUID deviceAnonymizedId = UUID.randomUUID();
+		String registrationToken = "token-" + UUID.randomUUID();
+		SystemMessage message = SystemMessage.createInstance("Hi there!");
```

Comment:
Those two lines can be extracted to a method `createTestMessage` or so

Reply:
Yes, sorry.